core/app/models/spree/shipping_rate.rb

```diff
@@ -1,46 +1,48 @@
 module Spree
   class ShippingRate < Spree::Base
+    attr_writer :order
```

Comment:
OK great. Looking at this was what prompted me to submit https://github.com/solidusio/solidus/pull/950, which I think is a good idea in any case, but might allow us to clean things up for this also.